odoo/tools/pdf/__init__.py

```diff
@@ -217,6 +217,24 @@ def to_pdf_stream(attachment) -> io.BytesIO:
     _logger.warning("mimetype (%s) not recognized for %s", attachment.mimetype, attachment)
 
 
+def extract_page(attachment, num_page=0) -> io.BytesIO | None:
+    """Exctract a specific page form an attachement pdf"""
+    pdf_stream = to_pdf_stream(attachment)
+    if not pdf_stream:
+        return
+    try:
+        pdf = PdfFileReader(pdf_stream)
+    except PdfReadError:
+        _logger.warning("Error reading pdf file %s", attachment)
+    else:
+        page = pdf.getPage(num_page)
+        pdf_writer = PdfFileWriter()
+        pdf_writer.addPage(page)
+        stream = io.BytesIO()
+        pdf_writer.write(stream)
+        return stream
```

Comment:
Shouldn't a "get page" function return the page instead?
We really need to re-create a pdf with just one page?